Sources/Workspace/Manifest.swift

```diff
@@ -0,0 +1,34 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2019 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import PackageModel
+import PackageLoading
+
+extension Manifest {
```

Comment:
It seems weird to attach this to the `Manifest`. This can be a static method on `ManifestLoader` instead.

Reply:
To be honest, it is very unintuitive that the types cannot construct themselves. It makes them hard to find. As an outsider, you typically look within the namespace of a type itself for ways to construct it. When I first started using SwiftPM, I only figured out how it was supposed to work by cloning and using Xcode to inspect the call hierarchy of the initializers in order to zoom out and see where the different pieces were coming from.

(I would have designed `ManifestLoader` as `internal` and only exposed it indirectly through a `public` extension to `Manifest`.)

I will move these extensions as you have asked, but maybe it is worth keeping the general thought in the back of your mind as you design going forward.

Reply:
This approach allows easier testing and mocking of these heavy weight operations like manifest loading, git operation, etc. It also makes sense to keep the loading mechanism decoupled with the model object itself. That said, we can improve the documentation on these types so users can figure out how to construct them without having to figure that out using a debugger.

Reply:
Moved this extension.

(Not pushed yet.)